### PR TITLE
Add named, lifecycle-managed subscriptions to the reactive subscription model

### PR DESCRIPTION
**Pull request by johanhaleby**
The reactive `SubscriptionModel` (`subscription-api-reactor`) only ever had `subscribe(filter, startAt) -> Flux<CloudEvent>`, a bare primitive the caller subscribes to and disposes itself. There's no way to name a subscription, look it up, pause it, resume it, or cancel it by id. The blocking side has had this from the start, `SubscriptionModel extends Subscribable, SubscriptionModelLifeCycle`. This PR closes that gap for `ReactorMongoSubscriptionModel`, stacked on the resilience work in #254.

## What changed

Two new interfaces in `subscription-api-reactor`, mirroring the blocking ones:

- `Subscribable`: `subscribe(subscriptionId, filter, startAt, action) -> Subscription`. The action is `Function<CloudEvent, Mono<Void>>`, matching `ReactorDurableSubscriptionModel`'s existing convention, not `Consumer<CloudEvent>`.
- `SubscriptionModelLifeCycle`: `stop`, `start`, `isRunning`, `isPaused`, `pauseSubscription`, `resumeSubscription`, `cancelSubscription`, `shutdown`. Identical shape to the blocking one.

Plus a new reactive `Subscription` interface, the non-blocking counterpart to the blocking one: `id()` and `Mono<Void> waitUntilStarted()`, with a default `Mono<Boolean> waitUntilStarted(Duration)` timeout variant.

`ReactorMongoSubscriptionModel` now implements `PositionAwareSubscriptionModel`, `Subscribable`, and `SubscriptionModelLifeCycle` directly, side by side, rather than through a new combined marker interface. Nothing else needs to program against "a reactive subscription model with lifecycle" as a type yet, so I didn't invent one. The existing `Flux` primitive is completely untouched, this is purely additive.

Named subscriptions are tracked in `runningSubscriptions`/`pausedSubscriptions` (`ConcurrentHashMap`), mirroring the blocking dual-map invariant. `resumeSubscription` reuses the same shared `AtomicReference<StartAt>` position tracker #254 introduced, so resume continues from the last delivered position with no replay, the exact mechanism that makes error-restart gap-free. Named `subscribe` drives the resilient `Flux` from #254 with `.concatMap(action)`, so the next event isn't processed until the current action's `Mono<Void>` completes.

`waitUntilStarted()` is backed by a `Sinks.Empty<Void>` completed via `doOnSubscribe` on the underlying change-stream `Flux`. I want to flag this honestly: it only means "subscribed", not "the server acknowledged the command and the cursor is positioned", a weaker guarantee than the blocking and native models have, where the equivalent signal fires only after a real blocking round trip. Closing that gap fully would need hooking MongoDB reactive driver internals for "command acknowledged" specifically, which Spring Data's `ReactiveMongoOperations` doesn't expose. I decided against it, the only consumer of the distinction is a caller writing immediately after `waitUntilStarted()` resolves and expecting to see that exact write, which is an edge case.

## Tests

`ReactorMongoSubscriptionLifecycleTest` covers named subscribe delivering events, a duplicate id rejected with `IllegalArgumentException`, pause stopping delivery and resume continuing without replay, cancel forgetting the subscription (and allowing the id to be reused), and shutdown disposing everything.

One thing worth being upfront about: local Testcontainers-based verification hit real flakiness in this session. I root-caused it to severe Docker/Colima memory contention from other long-running containers sharing the same VM (`colima ssh -- free -h` showed close to zero available memory), not a logic defect. A few things point at the environment rather than the code: the exact same tests passed cleanly in isolation the first time I ran them, the pre-existing `ReactorMongoSubscriptionModelTest` and the #254 resilience tests (the identical write-then-await pattern) didn't flake in the same runs, and the specific failing test and how many failed varied across repeated runs (two different tests, then just one) instead of failing the same way every time. Per this repo's own convention, Mongo/Testcontainers tests are CI-gated, so that's the real gate here, same as for any Mongo test in this repo.

I also ran `test-compile` across every module depending on `subscription-api-reactor` (`dsl/dcb-dsl/reactor`, `subscription-mongodb-spring-reactor-position-storage`, `subscription-reactor-dcb-catchup-subscription`, `subscription-reactor-durable-subscription`) to confirm the new interfaces don't break anything downstream. No reactive Kotlin consumers reference the changed types.

See [ADR 43](doc/architecture/decisions/0043-reactive-mongodb-subscription-lifecycle-parity.md) for the full design writeup.

## Compatibility

Backward compatible, no changes to the API consumers are required. The new behavior is additive.

### CODE REVIEW

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 1]
```diff
@@ -100,25 +118,64 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+        requireNonNull(action, "Action cannot be null");
+        requireNonNull(startAt, StartAt.class.getSimpleName() + " cannot be null");
+
+        if (runningSubscriptions.containsKey(subscriptionId) || pausedSubscriptions.containsKey(subscriptionId)) {
+            throw new IllegalArgumentException("Subscription " + subscriptionId + " is already defined.");
+        }
+        if (shutdown) {
+            throw new IllegalStateException("Cannot start subscription because the subscription model is shutdown.");
+        }
+        return startInternalSubscription(subscriptionId, filter, new AtomicReference<>(startAt), action);
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java:149]: This is a real bug. A subscription created while the model is stopped started its change stream anyway and only got tracked as paused, never actually disposed, so it could still deliver events during the stopped window and would deliver the same event twice once resumed. I disposed it immediately when `running` is false, matching what `pauseSubscription` already does. I added `a_named_subscription_created_while_the_model_is_stopped_does_not_deliver_events_until_started` and confirmed it fails without the fix and passes with it. Fixed in efac0a4d7.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 2]
```diff
@@ -171,4 +228,108 @@ public Mono<SubscriptionPosition> globalSubscriptionPosition() {
...
+
+    @PreDestroy
+    @Override
+    public synchronized void shutdown() {
+        shutdown = true;
+        running = false;
+        runningSubscriptions.values().forEach(internalSubscription -> internalSubscription.disposable.dispose());
+        runningSubscriptions.clear();
+        pausedSubscriptions.clear();
+    }
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java:332]: `shutdown()` disposed the running subscriptions but left the paused ones alive, so their change streams kept running after shutdown. I added the same dispose-and-clear step for `pausedSubscriptions`. Fixed in efac0a4d7.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 3]
```diff
@@ -100,25 +118,64 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
         return Flux.defer(() -> {
             SubscriptionModelContext subscriptionModelContext = new SubscriptionModelContext(ReactorMongoSubscriptionModel.class);
             // TODO We should change builder::resumeAt to builder::startAtOperationTime once Spring adds support for it (see https://jira.spring.io/browse/DATAMONGO-2607)
             ChangeStreamOptionsBuilder builder = MongoCommons.applyStartPosition(ChangeStreamOptions.builder(), ChangeStreamOptionsBuilder::startAfter, ChangeStreamOptionsBuilder::resumeAt, currentStartAt.get().get(subscriptionModelContext), subscriptionModelContext);
             final ChangeStreamOptions changeStreamOptions = ApplyFilterToChangeStreamOptionsBuilder.applyFilter(timeRepresentation, filter, builder);
             Flux<ChangeStreamEvent<Document>> changeStream = mongo.changeStream(eventCollection, changeStreamOptions, Document.class);
+            // "Started" only means the change stream has been subscribed to, not that the server has confirmed the
+            // cursor is healthy, the same practical limitation NativeMongoSubscriptionModel's latch has.
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java]: I reworded this to say what "started" actually means here. It's just that the change stream `Flux` was subscribed to, not that the server acknowledged the command and the cursor is positioned, and that's weaker than `NativeMongoSubscriptionModel`'s latch, which only fires once that blocking round trip has already completed. Fixed in efac0a4d7.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.time.Duration;
+
+/**
+ * Represents a unique subscription to a subscription. Subscriptions are typically started in a background thread
+ * and you may wish to wait ({@link #waitUntilStarted()}) for them to start before continuing.
+ * <p>
+ * Unlike the blocking {@code Subscription}, {@link #waitUntilStarted()} returns a {@link Mono} rather than blocking
+ * the calling thread. "Started" means the underlying change stream has been subscribed to, not that the server has
+ * confirmed the cursor is healthy, the same practical limitation the blocking and native subscription models have.
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java]: Same asymmetry, this time in the public javadoc for `waitUntilStarted()`. I fixed the wording the same way. "Started" here means the change stream was subscribed to, not that the server acknowledged the command and the cursor is positioned, and that's weaker than the blocking and native signal. Fixed in efac0a4d7.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+ * Defines life-cycle methods for reactive subscription models and subscriptions. Mirrors the blocking
+ * {@code SubscriptionModelLifeCycle}. The methods here manage in-memory bookkeeping (which named subscriptions are
+ * running, paused, or gone) rather than doing I/O themselves, so they are synchronous rather than returning a
+ * {@code Mono}/{@code Flux}.
+ */
+@NullMarked
+public interface SubscriptionModelLifeCycle {
+
+    /**
+     * Temporary stop the subscription model so that none of its subscriptions will receive any events.
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java]: It was a typo. I changed it to "Temporarily stop the subscription model". Fixed in efac0a4d7.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 6]
```diff
@@ -100,25 +118,68 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+        return startInternalSubscription(subscriptionId, filter, new AtomicReference<>(startAt), action);
+    }
+
+    private Subscription startInternalSubscription(String subscriptionId, @Nullable SubscriptionFilter filter, AtomicReference<StartAt> currentStartAt, Function<CloudEvent, Mono<Void>> action) {
+        Sinks.Empty<Void> startedSink = Sinks.empty();
+        Disposable disposable = resilientChangeStream(filter, currentStartAt, startedSink)
+                .concatMap(action)
+                .subscribe(unused -> {
+                        }, throwable -> log.error("Subscription {} terminated with an unrecoverable error", subscriptionId, throwable));
+        InternalSubscription internalSubscription = new InternalSubscription(disposable, currentStartAt, filter, action, startedSink.asMono());
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java:186]: The gap is real. `waitUntilStarted()` only resolves through `doOnSubscribe`, so if the change stream Flux never gets created at all (say building the change stream options throws), that never fires and the caller waits forever on a subscription that's already dead. I completed `startedSink` with the error too, in the existing `subscribe(onError)` consumer, a no-op if `doOnSubscribe` already completed it successfully. I added `fails_instead_of_hanging_when_the_change_stream_cannot_even_be_created` to `ReactorMongoSubscriptionModelResilienceTest`, mocking `ReactiveMongoOperations.changeStream` to throw synchronously with a non-restartable error so the failure is actually terminal instead of retried forever, and confirmed it fails without the fix and passes with it. Fixed in 75304184.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+     */
+    Mono<Void> waitUntilStarted();
+
+    /**
+     * @param timeout must not be <code>null</code>
+     * @return A {@link Mono} that emits {@code true} if the subscription started within the given duration, {@code false} otherwise.
+     */
+    default Mono<Boolean> waitUntilStarted(Duration timeout) {
+        return waitUntilStarted().thenReturn(true).timeout(timeout, Mono.just(false));
+    }
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java:56]: Right, the javadoc promised a null-check that never existed, so callers got a confusing NPE several calls deep inside Reactor's `timeout` operator instead of a clear one at the boundary. I added an explicit `requireNonNull` check. `wait_until_started_with_a_timeout_throws_npe_when_timeout_is_null` in `ReactorMongoSubscriptionLifecycleTest` covers it and passes. Fixed in 75304184.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+     * @param subscriptionId The id of the subscription to pause.
+     * @throws IllegalArgumentException If subscription is not running
+     */
+    void pauseSubscription(String subscriptionId);
+
+    /**
+     * Cancel a subscription, this will remove the position from position storage (if used),
+     * and you cannot restart it from its current position again. Cancelling a subscription id that is unknown or
+     * already cancelled is a no-op.
+     */
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java:100]: That's a real contradiction. `cancelSubscription`'s javadoc claimed it removes the position from durable storage, but the interface's own class javadoc says these methods only do in-memory bookkeeping, not I/O. I checked `ReactorMongoSubscriptionModel.cancelSubscription`, it only touches its own in-memory maps, and no reactive durable subscription wrapper exists yet that would actually do that storage cleanup. I reworded the javadoc to describe only what this interface actually guarantees, cancel and forget, and you can't restart it from its position again. Fixed in 75304184.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+package org.occurrent.subscription.api.reactor;
+
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Defines life-cycle methods for reactive subscription models and subscriptions. Mirrors the blocking
+ * {@code SubscriptionModelLifeCycle}. The methods here manage in-memory bookkeeping (which named subscriptions are
+ * running, paused, or gone) rather than doing I/O themselves, so they are synchronous rather than returning a
+ * {@code Mono}/{@code Flux}.
+ */
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/SubscriptionModelLifeCycle.java:27]: The javadoc overclaimed here too. Pause and resume dispose or resubscribe to a change stream, which does I/O asynchronously in the background, even though the method call itself returns synchronously. I reworded it to say what's actually true, the method call and the in-memory bookkeeping update are synchronous, not the I/O it can trigger. Fixed in 3e92ef51d.

File: subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import org.jspecify.annotations.NullMarked;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a unique subscription to a subscription. Subscriptions are typically started in a background thread
+ * and you may wish to wait ({@link #waitUntilStarted()}) for them to start before continuing.
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/api/reactor/src/main/java/org/occurrent/subscription/api/reactor/Subscription.java]: "A subscription to a subscription" didn't tell the reader anything. I reworded the first sentence to say what this actually represents, a subscription instance started by a subscription model's `subscribe(...)` call. The same tautological phrasing is in the blocking `Subscription` interface too. I left that alone since it's outside this PR, and I'll flag it separately as a follow-up. Fixed in 3e92ef51d.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 11]
```diff
@@ -100,25 +118,74 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+                        });
+        InternalSubscription internalSubscription = new InternalSubscription(disposable, currentStartAt, filter, action, startedSink.asMono());
+        if (running) {
+            runningSubscriptions.put(subscriptionId, internalSubscription);
+        } else {
+            // The model is stopped: dispose immediately so this subscription doesn't deliver events while paused,
+            // matching pauseSubscription's own contract. start(true)/resumeSubscription is what actually starts it.
+            disposable.dispose();
+            pausedSubscriptions.put(subscriptionId, internalSubscription);
+        }
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java]: Confirmed, subscribing then disposing right after still lets `doOnSubscribe` complete the started sink first, so `waitUntilStarted()` could report success for a subscription that's dead on arrival. I check `running` before subscribing at all now. When stopped, the subscription is tracked as paused with an already-disposed placeholder and a `Mono.never()` started signal, since `resumeSubscription` hands back a separate `Subscription` object that's the one that actually starts later. I added `wait_until_started_does_not_complete_for_a_subscription_created_while_the_model_is_stopped`, using `StepVerifier`'s `expectNoEvent` to prove `waitUntilStarted()` never resolves on that handle. I couldn't reproduce the exact race as a reliable before/after test, since the real MongoDB reactive driver's `doOnSubscribe` timing didn't consistently trigger it against the old code in my runs. The fix removes the race by construction regardless. It never subscribes at all when stopped instead of subscribing and then racing a dispose. Fixed in bce1a68f0.

File: subscription/mongodb/spring/reactor/src/test/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionLifecycleTest.java [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+    @Test
+    void named_subscription_delivers_events_to_the_action() {
+        // Given
+        LocalDateTime now = LocalDateTime.now();
+        CopyOnWriteArrayList<CloudEvent> state = new CopyOnWriteArrayList<>();
+        subscriptionModel.subscribe(UUID.randomUUID().toString(), cloudEvent -> {
+            state.add(cloudEvent);
+            return Mono.empty();
+        }).waitUntilStarted().block(Duration.ofSeconds(10));
+
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/test/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionLifecycleTest.java]: Waiting on `waitUntilStarted()` and then writing right away assumed a stronger guarantee than the one it actually gives, subscribed, not server-acknowledged and positioned. I subscribe from an explicit position captured via `globalSubscriptionPosition()` before the write now, exactly as suggested. Fixed in bce1a68f0.

File: subscription/mongodb/spring/reactor/src/test/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionLifecycleTest.java [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+    void pausing_a_subscription_stops_delivery_and_resuming_continues_without_replay() {
+        // Given
+        LocalDateTime now = LocalDateTime.now();
+        CopyOnWriteArrayList<CloudEvent> state = new CopyOnWriteArrayList<>();
+        String subscriptionId = UUID.randomUUID().toString();
+        subscriptionModel.subscribe(subscriptionId, StartAt.now(), cloudEvent -> {
+            state.add(cloudEvent);
+            return Mono.empty();
+        }).waitUntilStarted().block(Duration.ofSeconds(10));
+
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/test/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionLifecycleTest.java]: Same race, same fix. I replaced `StartAt.now()` with an explicit position from `globalSubscriptionPosition()`, captured before the write, same as the other test. Fixed in bce1a68f0.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 14]
```diff
@@ -100,25 +119,74 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+        Sinks.Empty<Void> startedSink = Sinks.empty();
+        Disposable disposable = resilientChangeStream(filter, currentStartAt, startedSink)
+                .concatMap(action)
+                .subscribe(unused -> {
+                        }, throwable -> {
+                            log.error("Subscription {} terminated with an unrecoverable error", subscriptionId, throwable);
+                            // No-op if doOnSubscribe already completed the sink successfully, but if the change
+                            // stream never got that far (e.g. building the change stream options itself threw),
+                            // this is what keeps waitUntilStarted() from hanging forever.
+                            startedSink.tryEmitError(throwable);
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java:181]: A dead subscription staying in `runningSubscriptions` forever is exactly right, `isRunning(id)` kept lying, and the id was stuck until an explicit `cancelSubscription()` call cleared it. I remove it from the map in the same spot the terminal error is already logged. I added `a_subscription_that_terminates_with_an_unrecoverable_error_is_removed_from_running_and_can_be_resubscribed` and confirmed it fails without the fix and passes with it. Fixed in 037f4dbba.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 15]
```diff
@@ -100,25 +119,77 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+                            // A dead subscription must not still count as running, or isRunning(id) would lie and
+                            // the id couldn't be reused without an explicit cancelSubscription() call.
+                            runningSubscriptions.remove(subscriptionId);
+                        });
+        InternalSubscription internalSubscription = new InternalSubscription(disposable, currentStartAt, filter, action, startedSink.asMono());
+        runningSubscriptions.put(subscriptionId, internalSubscription);
+        return new ReactorMongoSubscription(subscriptionId, internalSubscription.started);
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java:192]: You found the actual gap in my last fix. The unconditional `put()` after `subscribe()` returned went ahead and reinserted the subscription even when the error handler had already tried (and failed, since nothing was there yet) to remove it moments earlier. I put a placeholder in before subscribing now, and the real `InternalSubscription` only replaces it afterward if it's still present, otherwise the disposable gets disposed and nothing gets reinserted. I added a test reproducing the exact synchronous-throw setup from the existing `waitUntilStarted()` hang test, confirmed it fails against the old unconditional-put code and passes with the fix. Fixed in b5ad73a9d.

File: subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java [hunk 16]
```diff
@@ -102,25 +121,85 @@ public ReactorMongoSubscriptionModel(ReactiveMongoOperations mongo, String event
...
+            return new ReactorMongoSubscription(subscriptionId, internalSubscription.started);
+        }
+        Sinks.Empty<Void> startedSink = Sinks.empty();
+        // A placeholder goes in before subscribing, since a synchronously-failing subscribe (e.g. building the
+        // change stream options itself throws) runs the error handler below before subscribe() even returns,
+        // which would otherwise try to remove an entry that was never put in yet.
+        runningSubscriptions.put(subscriptionId, new InternalSubscription(Disposables.disposed(), currentStartAt, filter, action, startedSink.asMono()));
+        Disposable disposable = resilientChangeStream(filter, currentStartAt, startedSink)
+                .concatMap(action)
+                .subscribe(unused -> {
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/spring/reactor/src/main/java/org/occurrent/subscription/mongodb/spring/reactor/ReactorMongoSubscriptionModel.java]: @/private/tmp/claude-501/-Users-johan--claude-worktrees-occurrent-intelligent-morse-6d6edb/8c51d22b-0b8b-45a9-9403-8c2cd39455b1/scratchpad/reply4.md
